lib/resque/scheduler.rb

```diff
@@ -59,10 +59,8 @@ def run
           # Now start the scheduling part of the loop.
           loop do
             begin
-              if master?
-                handle_delayed_items
-                update_schedule if dynamic
-              end
+              handle_delayed_items
+              update_schedule if dynamic && master?
             rescue Errno::EAGAIN, Errno::ECONNRESET, Redis::CannotConnectError => e
               log! e.message
               release_master_lock
```

Comment:
Hm. This looks a bit odd. It reads like we might potentially release the master lock even though we never checked if we were actually master.

Reply:
This does check before releasing but I make this clear in https://github.com/Shopify/resque-scheduler/pull/26